cmake/vecmem-compiler-options-cuda.cmake

```diff
@@ -11,7 +11,7 @@ include_guard( GLOBAL )
 include( vecmem-functions )
 
 # Set up the used C++ standard(s).
-set( CMAKE_CUDA_STANDARD 14 CACHE STRING "The (CUDA) C++ standard to use" )
+set( CMAKE_CUDA_STANDARD 17 CACHE STRING "The (CUDA) C++ standard to use" )
```

Comment:
I hope this is not needed, otherwise we'll be in trouble.

To be able to work with SYCL on NVidia devices, we need to keep using CUDA 10.X. Which does not allow for C\+\+17. :frowning:

Reply:
That's a little awkward. Much of the code relies on C++17 features, so if we are to use C++17 for all of the standard C++ compilation and C++14 for the CUDA code, we would need to define a really clear split between the headers of the two... :disappointed:

Reply:
:thinking: It should not be too bad. `vecmem::cuda::device_resource` (and friends) will need to be a C\+\+ class, with a `.cpp` implementation file. Then internally the CUDA library can create helper functions for allocating memory on a GPU. These helper functions can be implemented in `.cu` files, compiled with C\+\+14. It's really just one degree of separation that we need.